### PR TITLE
Analytics tab

### PR DESCRIPTION
**Pull request by taiwoAkindele**
Analytics tabs on the developers page

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -49,6 +52,7 @@ const App:React.FC = () => {
       dispatch(login(user))
     }
     loginUser()
+    dispatch(getApis())
```
Comment (pablo-clueless) [inline src/App.tsx]: Create another effect for this action dispatch and lodge it in a `useMemo`, we will pass dependencies to it later.

File: src/redux/slices/apiSlice.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ const initialState = {
 
 export const getApis = createAsyncThunk("apis/getApis", async(_, thunkAPI) => {
     try {
-        const response = await fetch(`${url}/categories`)
+        const response = await fetch(`${url}/zl-core/categories`)
```
Comment (pablo-clueless) [inline src/redux/slices/apiSlice.ts]: We are appending only the routes here, so `/categories` only

File: src/redux/slices/apiSlice.ts [hunk 3]
```diff
@@ -38,8 +38,8 @@ const apiSlice = createSlice({
         builder.addCase(getApis.pending, (state) => {
             state.loading = "pending"
         }),
-        builder.addCase(getApis.fulfilled, (state, action: PayloadAction<any>) => {
-            state.apis.push(action.payload)
+        builder.addCase(getApis.fulfilled, (state, { payload }) => {
+            state.apis = payload
```
Comment (pablo-clueless) [inline src/redux/slices/apiSlice.ts:42]: Why the change??

**Review by pablo-clueless — CHANGES_REQUESTED** [review]
Check through for comments

File: src/redux/slices/apiSlice.ts (hunk 3 above)
Comment (taiwoAkindele) [reply src/redux/slices/apiSlice.ts:42]: It is creating another array to store the categories instead of storing it in the initial array.

File: src/redux/slices/apiSlice.ts (hunk 2 above)
Comment (taiwoAkindele) [reply src/redux/slices/apiSlice.ts]: alright

File: src/redux/slices/apiSlice.ts (hunk 3 above)
Comment (pablo-clueless) [reply src/redux/slices/apiSlice.ts:42]: I'll check it later.

**Review by pablo-clueless — APPROVED** [review]
